src/plugins/decor/index.tsx

```diff
@@ -0,0 +1,195 @@
+        {
+            find: "getAvatarDecorationURL:",
+            replacement: {
+                match: /(function \i\(\i\){)(let{avatarDecoration)/,
+                replace: "$1const vcDecorDecoration=$self.getDecorAvatarDecorationURL(arguments[0]);if(vcDecorDecoration)return vcDecorDecoration;$2"
+            }
+        },
+        // Patch profile customization settings to include Decor section
+        {
+            find: "DefaultCustomizationSections",
+            replacement: {
+                match: /{user:\i},"decoration"\),/,
+                replace: "$&$self.DecorSection(),"
+            }
+        },
+        // Decoration modal module
+        {
+            find: ".decorationGridItem",
+            replacement: [{
+                match: /(\i=)(\i=>{let{children:\i)/,
+                replace: "$1$self.DecorationGridItem=$2"
+            },
+            {
+                match: /(\i=)(\i=>{let{user:\i,avatarDecoration)/,
+                replace: "$1$self.DecorationGridDecoration=$2"
+            },
+            {
+                match: /(\i===\i\.Section\.PURCHASE\|\|\i===\i\.Section\.PREMIUM_PURCHASE&&\i)(?<=avatarDecoration:(\i).{0,800}?)/,
+                replace: "$2.skuId === $self.SKU_ID || ($1)"
+            }]
+        }
+    ],
+
+    flux: {
+        CONNECTION_OPEN: () => {
+            useAuthorizationStore.getState().init();
+            useCurrentUserDecorationsStore.getState().clear();
+            useUsersDecorationsStore.getState().fetch(UserStore.getCurrentUser().id, true);
+        },
+        USER_PROFILE_MODAL_OPEN: data => {
+            useUsersDecorationsStore.getState().fetch(data.userId, true);
+        },
+        MESSAGE_CREATE: data => {
+            const channelId = SelectedChannelStore.getChannelId();
+            if (data.channelId === channelId) {
+                useUsersDecorationsStore.getState().fetch(data.message.author.id);
+            }
+        },
+        TYPING_START: data => {
+            const channelId = SelectedChannelStore.getChannelId();
+            if (data.channelId === channelId) {
+                useUsersDecorationsStore.getState().fetch(data.userId);
+            }
+        },
+        LOAD_MESSAGES_SUCCESS: ({ messages }) => {
+            useUsersDecorationsStore.getState().fetchMany(messages.map(m => m.author.id));
+        },
+        // TODO: Still need to fetch for member list
+    },
+
+    set DecorationGridItem(e: any) {
+        setDecorationGridItem(e);
+    },
+
+    set DecorationGridDecoration(e: any) {
+        setDecorationGridDecoration(e);
+    },
+
+    SKU_ID,
+
+    async start() {
+        useUsersDecorationsStore.getState().fetch(UserStore.getCurrentUser().id, true);
+    },
+
+    patchUser(user) {
+        if (!user) return user;
+
+        const store = useUsersDecorationsStore.getState();
+
+        // No decor fetched yet
+        if (!store.has(user.id)) return user;
+
+        const decoration = store.get(user.id);
```

Comment:
```suggestion
        const decoration = store.get(user.id);
        if (!decoration) return user;
```
and remove the above if (!store.has())

Reply:
fine